### PR TITLE
enh(Centreon) :  rework the header for it to be more responsive

### PR DESCRIPTION
**Pull request by Med1233**
## Description

The main purpose of this PR is to enhance the header styling and add some extra features to it. In addition to that, we make sure to make the header responsive using different designs according to differents breakpoints 

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie
- develop 

<h2> How this pull request can be tested ? </h2>

this can be tested with additional modules : centreon-frontend and centreon-bam 

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/Header/Clock/index.tsx [hunk 1]
```diff
@@ -49,8 +53,13 @@ const Clock = (): JSX.Element => {
 
   return (
     <div className={classes.dateTime} data-cy="clock">
-      <Typography variant="body2">{date}</Typography>
-      <Typography variant="body1">{time}</Typography>
+      <Typography style={{ lineHeight: '1' }} variant="body2">
```
Comment (Thebarda) [inline www/front_src/src/Header/Clock/index.tsx]: It's better to use className

File: www/front_src/src/Header/PollerMenu/ExportConfiguration/index.tsx [hunk 2]
```diff
@@ -30,7 +30,14 @@ interface Props {
 
 const useStyles = makeStyles((theme) => ({
   exportButton: {
+    '&:hover': {
+      background: '#969faf',
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/ExportConfiguration/index.tsx]: Please use a color from the MUI theme palette

File: www/front_src/src/Header/PollerMenu/ExportConfiguration/index.tsx [hunk 3]
```diff
@@ -30,7 +30,14 @@ interface Props {
...
+    },
+    background: theme.palette.common.black,
+    border: '1px solid white',
+    color: theme.palette.common.white,
     display: 'flex',
+    fontSize: theme.spacing(1.4),
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/ExportConfiguration/index.tsx]: Use a fontSize from the theme

File: www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx [hunk 4]
```diff
@@ -50,13 +50,35 @@ const useStatusStyles = makeStyles<Theme, StyleProps>((theme) => {
...
+        width: theme.spacing(1.8),
+      },
+    },
+    container: {
+      display: 'flex',
+      gap: theme.spacing(0.5),
+      marginLeft: theme.spacing(-0.5),
+      [theme.breakpoints.down(768)]: {
+        gap: theme.spacing(0.4),
+        marginLeft: theme.spacing(-1.2),
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx]: Do not use negative margins

File: www/front_src/src/Header/PollerMenu/index.tsx [hunk 5]
```diff
@@ -49,19 +49,53 @@ interface PollerData {
 
 const useStyles = makeStyles((theme) => ({
   confButton: {
+    '&:hover': {
+      background: '#969faf',
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/index.tsx]: Please use a color from the MUI theme palette

File: www/front_src/src/Header/PollerMenu/index.tsx [hunk 6]
```diff
@@ -49,19 +49,53 @@ interface PollerData {
...
   container: {
+    borderRight: '1px solid white',
+    display: 'flex',
+    paddingRight: theme.spacing(3),
+    // [theme.breakpoints.down(1200)]: {
+    //   paddingRight: theme.spacing(1.5),
+    // },
+    [theme.breakpoints.down(768)]: {
+      // paddingRight: theme.spacing(0.8),
+      paddingRight: '10px',
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/index.tsx]: Use spacing from the theme

File: www/front_src/src/Header/PollerMenu/index.tsx [hunk 7]
```diff
@@ -49,19 +49,53 @@ interface PollerData {
...
+    },
+  },
+  iconToggleMenu: {
+    alignItems: 'flex-end',
     display: 'flex',
-    flex: 0.4,
+    [theme.breakpoints.down(768)]: {
+      alignItems: 'center',
+      justifyContent: 'center',
+      marginLeft: '-5px',
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/index.tsx]: Use spacing from the theme
Do not use negative margins

File: www/front_src/src/Header/RessourceStatusCounter/index.tsx [hunk 8]
```diff
@@ -14,18 +14,53 @@ export const useStyles = makeStyles((theme) => ({
...
     textAlign: 'left',
     top: '100%',
-    width: '100%',
+    width: '135%',
+    [theme.breakpoints.down(768)]: {
+      width: '170%',
+    },
     zIndex: theme.zIndex.mobileStepper,
   },
+  subMenuToggle_extraWidth: {
```
Comment (Thebarda) [inline www/front_src/src/Header/RessourceStatusCounter/index.tsx]: Do not write using snake case

File: www/front_src/src/Header/RessourceStatusCounter/index.tsx [hunk 9]
```diff
@@ -14,18 +14,53 @@ export const useStyles = makeStyles((theme) => ({
...
+    width: '170%',
```
Comment (Thebarda) [inline www/front_src/src/Header/RessourceStatusCounter/index.tsx]: Why 170%?

**Review by Thebarda — COMMENTED** [review]
Remove comments if not necessary.

I think it's better to refactorize Host and Service into one re-usable component. So we can comsume it in BAM (and other modules)

**Comment by bdauria** [comment]
@Med1233 you have to re-generate the Header visual tests for the build to pass

File: www/front_src/src/Header/RessourceStatusCounter/index.tsx (hunk 9 above)
Comment (Med1233) [reply www/front_src/src/Header/RessourceStatusCounter/index.tsx]: From a specific breakpoint, the submenu content starts to overflow. So I have to increase the menu width to keep each "li" shown clearly in one line.

File: www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx [hunk 10]
```diff
@@ -50,13 +50,36 @@ const useStatusStyles = makeStyles<Theme, StyleProps>((theme) => {
...
-      fontSize: theme.typography.body1.fontSize,
-      height: theme.spacing(4),
-      position: 'relative',
-      width: theme.spacing(4),
+      height: theme.spacing(1.5),
+      width: theme.spacing(1.5),
+      [theme.breakpoints.down(768)]: {
+        height: theme.spacing(1.3),
+        width: theme.spacing(1.3),
+      },
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx]: Use Integer if possible

File: www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx [hunk 11]
```diff
@@ -50,13 +50,36 @@ const useStatusStyles = makeStyles<Theme, StyleProps>((theme) => {
...
+    container: {
+      display: 'flex',
+      gap: theme.spacing(0.5),
+      position: 'relative',
+      right: theme.spacing(0.5),
+      [theme.breakpoints.down(768)]: {
+        gap: theme.spacing(0.4),
+        right: theme.spacing(1),
+        top: theme.spacing(0.4),
+      },
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx]: Are you trying to center an element here?

File: www/front_src/src/Header/PollerMenu/index.tsx [hunk 12]
```diff
@@ -49,19 +48,51 @@ interface PollerData {
...
     display: 'flex',
+    fontSize: theme.typography.body2.fontSize,
     marginTop: theme.spacing(1),
   },
   container: {
+    borderRight: '1px solid white',
+    display: 'flex',
+    paddingRight: theme.spacing(3),
+    [theme.breakpoints.down(768)]: {
+      paddingRight: theme.spacing(1.25),
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/index.tsx]:
```suggestion
      paddingRight: theme.spacing(1),
```

File: www/front_src/src/Header/PollerMenu/index.tsx [hunk 13]
```diff
@@ -49,19 +48,51 @@ interface PollerData {
...
+  },
+  iconToggleMenu: {
+    alignItems: 'flex-end',
     display: 'flex',
-    flex: 0.4,
+    [theme.breakpoints.down(768)]: {
+      alignItems: 'center',
+      justifyContent: 'center',
+      position: 'relative',
+      right: theme.spacing(0.6),
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/index.tsx]:
```suggestion
      right: theme.spacing(1),
```

File: www/front_src/src/Header/index.tsx [hunk 14]
```diff
@@ -13,43 +13,32 @@ const useStyles = makeStyles((theme) => ({
   container: {
     alignItems: 'center',
     display: 'flex',
+    justifyContent: 'space-between',
+    margin: '0 34px 0 24px',
```
Comment (Thebarda) [inline www/front_src/src/Header/index.tsx]: Use theme spacing

File: www/front_src/src/Header/index.tsx [hunk 15]
```diff
@@ -13,43 +13,32 @@ const useStyles = makeStyles((theme) => ({
...
-    justifyContent: 'flex-end',
-  },
-  hostStatusContainer: {
     display: 'flex',
-    flex: 0.35,
-    justifyContent: 'center',
-  },
-  pollerContainer: {
-    flex: 0.4,
+    height: '70px',
```
Comment (Thebarda) [inline www/front_src/src/Header/index.tsx]: same comment as above

File: www/front_src/src/Header/index.tsx [hunk 16]
```diff
@@ -13,43 +13,32 @@ const useStyles = makeStyles((theme) => ({
...
     alignItems: 'center',
     display: 'flex',
-    flex: 0.9,
-  },
-  serviceStatusContainer: {
-    display: 'flex',
-    flex: 0.45,
+    gap: theme.spacing(3),
+    [theme.breakpoints.down(600)]: {
+      gap: '10px',
```
Comment (Thebarda) [inline www/front_src/src/Header/index.tsx]: same comment as above

File: www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx (hunk 11 above)
Comment (Med1233) [reply www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx]: No, I translated the small icons near to the big icon

File: www/front_src/src/Header/userMenu/index.tsx [hunk 17]
```diff
@@ -346,8 +394,21 @@ const UserMenu = ({ headerRef }: Props): JSX.Element => {
...
+            <SearchIcon className={classes.searchIcon} />
+          </Badge> */}
+          {/* 
+          <Badge
+            badgeContent={4}
+            classes={{ badge: classes.badge }}
+            color="error"
+          >
+            <NotificationsNoneIcon className={classes.notificationIcon} />
+          </Badge> */}
```
Comment (bdauria) [inline www/front_src/src/Header/userMenu/index.tsx]:
```suggestion
```

File: www/front_src/src/Header/userMenu/index.tsx [hunk 18]
```diff
@@ -151,21 +187,33 @@ const useStyles = makeStyles((theme) => ({
...
     cursor: 'pointer',
-    marginLeft: theme.spacing(1),
+    fontSize: theme.spacing(3.7),
   },
+  // userIconBadge: {
+  //   border: '1.5px solid white',
+  //   borderRadius: '100%',
+  //   cursor: 'pointer',
+  //   padding: theme.spacing(0.4),
+  // },
```
Comment (bdauria) [inline www/front_src/src/Header/userMenu/index.tsx]:
```suggestion
```

File: www/front_src/src/Header/userMenu/index.tsx [hunk 19]
```diff
@@ -133,13 +156,26 @@ const useStyles = makeStyles((theme) => ({
...
     zIndex: theme.zIndex.tooltip,
   },
+  searchIcon: {
+    color: theme.palette.common.white,
+    cursor: 'pointer',
+    fontSize: theme.spacing(3.3),
+    [theme.breakpoints.down(600)]: {
+      display: 'none',
+    },
+  },
```
Comment (bdauria) [inline www/front_src/src/Header/userMenu/index.tsx]:
```suggestion
```

File: www/front_src/src/Header/userMenu/index.tsx [hunk 20]
```diff
@@ -133,13 +156,26 @@ const useStyles = makeStyles((theme) => ({
   nameContainer: {
     padding: theme.spacing(0, 2, 0.25, 2.25),
   },
+  notificationIcon: {
+    color: theme.palette.common.white,
+    cursor: 'pointer',
+    fontSize: theme.spacing(3.3),
+  },
```
Comment (bdauria) [inline www/front_src/src/Header/userMenu/index.tsx]:
```suggestion
```

File: www/front_src/src/Header/userMenu/index.tsx [hunk 21]
```diff
@@ -21,7 +21,8 @@ import {
   ListItemIcon as MUIListItemIcon,
   Fade,
 } from '@mui/material';
-import UserIcon from '@mui/icons-material/AccountCircle';
+// import UserIcon from '@mui/icons-material/AccountCircle';
```
Comment (bdauria) [inline www/front_src/src/Header/userMenu/index.tsx]:
```suggestion
```

File: www/front_src/src/Header/Clock/index.tsx [hunk 22]
```diff
@@ -6,8 +6,15 @@ import makeStyles from '@mui/styles/makeStyles';
 import { centreonUi } from '../helpers/index';
 
 const useStyles = makeStyles((theme) => ({
+  date: {
+    whiteSpace: 'nowrap',
+  },
   dateTime: {
```
Comment (bdauria) [inline www/front_src/src/Header/Clock/index.tsx:9]: This style doesn't seem to be used

**Pull request closed without merging**